src/InvestmentManager.sol

```diff
@@ -245,6 +245,7 @@ contract InvestmentManager is Auth, IInvestmentManager {
         uint128 shares
     ) public auth {
         address vault = poolManager.getVault(poolId, trancheId, assetId);
+        require(vault != address(0), "InvestmentManager/vault-not-found");
```

Comment:
I think we can add this check to `PoolManager.getVault`, similar to what we do in `PoolManager.getVaultAsset`